local-cloudbuild.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env python
+#
+# Run the steps of cloudbuild.yaml locally, emulating Google Container
+# Builder functionality.  Does not actually push images to the
+# Container Registry.  Not all functionality is supported.
+#
+# Based on https://cloud.google.com/container-builder/docs/api/build-steps
+
+# System packages
+import argparse
+import getpass
+import os
+import shutil
+import subprocess
+import sys
+import tempfile
+
+# Third party packages
+import yaml
+
+
+# Types
+class CloudBuildError(Exception):
+    pass
+
+
+def main(argv):
+    """Main entrypoint for cli"""
+    parser = argparse.ArgumentParser(
+        description='Process cloudbuild.yaml locally to build Docker images')
+    parser.add_argument(
+        'cloudbuild', type=str, help='Path to cloudbuild.yaml input file')
+    parser.add_argument(
```

Comment:
I couldn't think of any case where I actually wanted to specify it, and I can't just put it inside this directory tree (which I wanted to), because then it gets uploaded by container builder and causes weird circularities and name clashes.